pkg/volume/util/util.go

```diff
@@ -571,6 +571,38 @@ func IsLocalEphemeralVolume(volume v1.Volume) bool {
 		volume.ConfigMap != nil
 }
 
+// GetPersistentVolumeNodeNames returns the PersistentVolume node affinity
+// node name(s). nil is returned if the PV does not have the
+// specific node affinity node selector terms and match expressions.
+// PersistentVolume with node affinity has select and match expressions
+// in the form of:
+// nodeAffinity:
+//   required:
+//     nodeSelectorTerms:
+//     - matchExpressions:
+//       - key: kubernetes.io/hostname
+//         operator: In
+//         values:
+//         - <node1>
+//         - <node2>
+func GetPersistentVolumeNodeNames(pv *v1.PersistentVolume) []string {
+	if pv == nil || pv.Spec.NodeAffinity == nil || pv.Spec.NodeAffinity.Required == nil {
+		return nil
+	}
+	if len(pv.Spec.NodeAffinity.Required.NodeSelectorTerms) != 1 {
+		return nil
+	}
+	nodeSelector := pv.Spec.NodeAffinity.Required.NodeSelectorTerms[0]
+	if len(nodeSelector.MatchExpressions) != 1 {
+		return nil
+	}
+	matchExpr := nodeSelector.MatchExpressions[0]
+	if matchExpr.Key != v1.LabelHostname || matchExpr.Operator != v1.NodeSelectorOpIn {
```

Comment:
So we want this to be an optimization only for local volumes? Maybe I would call this function `GetLocalPersistentVolumeNodeNames` instead.

Reply:
Done, renamed to `GetLocalPersistentVolumeNodeNames`.